workflow/controller/operator.go

```diff
@@ -472,27 +483,23 @@ func (woc *wfOperationCtx) podReconciliation() error {
 		}
 	}
 
+	parallelPodNum := make(chan string, 500)
+	var wg sync.WaitGroup
 	for _, pod := range podList.Items {
-		origNodeStatus := *woc.wf.Status.DeepCopy()
-		performAssessment(&pod)
-		err = woc.applyExecutionControl(&pod)
-		if err != nil {
-			woc.log.Warnf("Failed to apply execution control to pod %s", pod.Name)
-		}
-		err = woc.checkAndCompress()
```

Comment:
@jessesuen Not yes.   I think the code `err = woc.checkAndCompress()` here is useless.   The function `performAssessment ` here is to make every pod state sync to nodeState.  It needn't to be compress.

All workflow will be pass to `persistUpdates` function, in that function,  it will be `checkAndCompress`. So I think this line can be removed.   I have test compress function in my prod env,  and it works without this line.  WDYT.

And add @sarabala1979  to double check this.

Reply:
`err = woc.checkAndCompress()` is required in podReconciliation. As per the existing call flow, every Node will update the nodestatus and output in podReconciliation function.  Node Status compress feature needs to check every nodes update. 
`err = woc.checkAndCompress()`  will execute only size check until it reach the size limit.

Cons:
1. This will fail only the node which is crossing the limit.

Reply:
Yep,  I have already put `checkAndCompress ` back.